src/app/data-client.test.ts

```diff
@@ -0,0 +1,263 @@
+import { FakeTransportBuilder } from '@improbable-eng/grpc-web-fake-transport';
+import { Timestamp } from 'google-protobuf/google/protobuf/timestamp_pb';
+import { Struct } from 'google-protobuf/google/protobuf/struct_pb';
+import {
+  beforeEach,
+  describe,
+  expect,
+  type SpyInstance,
+  test,
+  vi,
+} from 'vitest';
+import {
+  BinaryData,
+  BinaryDataByFilterRequest,
+  BinaryDataByFilterResponse,
+  BinaryDataByIDsRequest,
+  BinaryID,
+  CaptureInterval,
+  DataRequest,
+  Filter,
+  TabularData,
+  TabularDataByFilterRequest,
+  TabularDataByFilterResponse,
+  TagsFilter,
+} from '../gen/app/data/v1/data_pb';
+import { DataServiceClient } from '../gen/app/data/v1/data_pb_service';
+vi.mock('../gen/app/data/v1/data_pb_service');
+import { DataClient } from './data-client';
+
+const subject = () =>
+  new DataClient('fakeServiceHost', {
+    transport: new FakeTransportBuilder().build(),
+  });
+describe('DataClient tests', () => {
+  describe('tabularDataByFilter tests', () => {
+    let methodSpy: SpyInstance;
+    const tabData1 = new TabularData();
+    const tabData2 = new TabularData();
+    tabData1.setData(Struct.fromJavaScript({ key: 'value1' }));
+    tabData2.setData(Struct.fromJavaScript({ key: 'value2' }));
+    const tabDataResponse = new TabularDataByFilterResponse();
+    tabDataResponse.setDataList([tabData1, tabData2]);
+
+    beforeEach(() => {
+      methodSpy = vi
+        .spyOn(DataServiceClient.prototype, 'tabularDataByFilter')
+        // @ts-expect-error compiler is matching incorrect function signature
+        .mockImplementationOnce((_req: TabularDataByFilterRequest, _md, cb) => {
+          cb(null, tabDataResponse);
+        })
+        // @ts-expect-error compiler is matching incorrect function signature
+        .mockImplementation((_req, _md, cb) => {
+          cb(null, {
+            getDataList: () => [],
+          });
+        });
+    });
+
+    test('get tabular data', async () => {
+      const promise = await subject().tabularDataByFilter();
+      expect(promise.length).toEqual(2);
+      const [data1, data2] = promise;
+      expect(data1).toMatchObject(tabData1.toObject());
+      expect(data2).toMatchObject(tabData2.toObject());
+    });
+
+    test('get filtered tabular data', async () => {
+      const filter = subject().createFilter({
+        componentName: 'testComponentName',
+        componentType: 'testComponentType',
+      });
+
+      const dataReq = new DataRequest();
+      dataReq.setFilter(filter);
+      dataReq.setLimit(100);
+      dataReq.setLast('');
+      const req = new TabularDataByFilterRequest();
+      req.setDataRequest(dataReq);
+      req.setCountOnly(false);
+
+      await subject().tabularDataByFilter(filter);
+      expect(methodSpy).toHaveBeenCalledWith(
+        req,
```

Comment:
How do you feel about a more verbose name for the expected request to match the internal call to the data service client?

``suggestion
      const expectedRequest = new TabularDataByFilterRequest();
      expectedRequest.setDataRequest(dataReq);
      expectedRequest.setCountOnly(false);

      await subject().tabularDataByFilter(filter);
      expect(methodSpy).toHaveBeenCalledWith(
        expectedRequest,
```

Reply:
I was just following how it was in the data client. I don't really have an opinion on this, so I'll make this change